### DISCUSSION TITLE
Speaker diarization library I made

### DISCUSSION THREAD

**Discussion by thewh1teagle** [ggml-org/whisper.cpp#2430] [Show and tell]
Recently I created a [library](https://crates.io/crates/pyannote-rs) for speaker diarization. 
I also integrated it into the desktop app [Vibe](https://github.com/thewh1teagle/vibe)
The segmentation works pretty good, as well the speaker recognition.
Rust library can provide C-API. so we may be able to integrate it into whisper.cpp in future as well!

**Comment by stevevaius2015** [comment]
Could you elaborate more as I do not know Rust but need to diarize many audio files in my hobby work (parliament speeches) How to test it?
